src/PetriEngine/Colored/Reduction/RedRuleRelevance.cpp

```diff
@@ -0,0 +1,129 @@
+                            for (; prtIn != potentiallyRelevantTrans.input_arcs.end(); ++prtIn){
+                                if (prtIn->place >= arc.place) break;
+                            }
+                            if (prtIn != potentiallyRelevantTrans.input_arcs.end() && prtIn->place == arc.place) {
+                                const auto& prtOut = red.getOutArc(potentiallyRelevantTrans, arc.place);
+                                if (const auto ms1 = PetriEngine::Colored::extractVarMultiset(*prtIn->expr)){
+                                    if (const auto ms2 = PetriEngine::Colored::extractVarMultiset(*prtIn->expr)) {
+                                        if (ms1 == ms2){
+                                            continue;
+                                        }
+                                    }
+                                }
+                            }
+                            tseen[prtID] = true;
+                            wtrans.push_back(prtID);
+                        }
+                    }
+                } else {
+                    for (const auto prtID : place._post) {
+                        if (!tseen[prtID]) {
+                            // Summary of block: the potentially relevant transition is seen unless it:
+                            // - Is inhibited by 'place'
+                            // - Forms a decreasing loop on 'place' that cannot lower the token count of 'place' below the weight of 'arc'
+                            // - Forms a non-decreasing loop on 'place'
+                            const Transition &potentiallyRelevantTrans = red.transitions()[prtID];
+                            auto prtOut = potentiallyRelevantTrans.output_arcs.begin();
+                            for (; prtOut != potentiallyRelevantTrans.output_arcs.end(); ++prtOut)
+                                if (prtOut->place >= arc.place) break;
+
+                            if (prtOut != potentiallyRelevantTrans.output_arcs.end() && prtOut->place == arc.place) {
+                                const auto prtIn = red.getInArc(arc.place, potentiallyRelevantTrans);
+                                if (prtIn->inhib_weight > 0 || prtOut->expr->weight() >= arc.expr->weight() ||
+                                        prtOut->expr->weight() >= prtIn->expr->weight())
+                                    continue;
+                            }
+                            tseen[prtID] = true;
+                            wtrans.push_back(prtID);
+                        }
+                    }
+                }
+            }
+        }
+
+        for (uint32_t i = red.placeCount(); i > 0;) {
+            // This loop structure avoids underflow handling while also minimizing placeCount() calls.
+            i--;
+            if (pseen[i] || red.places()[i].skipped){
+                continue;
+            } else {
+                red.skipPlace(i);
+                changed |= true;
```

Comment:
Fancy, men dette er nok
```suggestion
                changed = true;
```